src/samples/coap/lwm2m-client.c

```diff
@@ -623,22 +633,47 @@ main(int argc, char *argv[])
         goto exit;
     }
 
+    security_data = calloc(1, sizeof(struct security_obj_instance_ctx));
+    if (!security_data) {
+        fprintf(stderr, "Could not alloc memory for security object context\n");
+        return -ENOMEM;
+    }
+
+    security_data->client = client;
+
     if (!data_ctx.is_bootstrap) {
         r = sol_lwm2m_add_object_instance(client, &server_object, NULL);
+
         if (r < 0) {
             fprintf(stderr, "Could not add a server object instance\n");
             goto exit_del;
         }
+/*
+        security_data->server_uri = (char*)malloc((strlen("coap://localhost:5683") + 1) * sizeof(char));
+        strcpy(security_data->server_uri, "coap://localhost:5683");
+ */
+        security_data->server_uri = sol_str_slice_to_str(sol_str_slice_from_blob(&server_addr));
+        security_data->is_bootstrap = false;
+        security_data->server_id = 101;
     } else {
         r = sol_lwm2m_client_add_bootstrap_finish_monitor(client, bootstrap_cb,
             NULL);
+
         if (r < 0) {
             fprintf(stderr, "Could not add a bootstrap monitor\n");
             goto exit_del;
         }
+/*
+        security_data->server_uri = (char*)malloc((strlen("coap://localhost:5783") + 1) * sizeof(char));
+        strcpy(security_data->server_uri, "coap://localhost:5783");
+ */
+        security_data->server_uri = sol_str_slice_to_str(sol_str_slice_from_blob(&bootstrap_server_addr));
```

Comment:
ditto. The `bootstrap_server_addr` is statically defined in this same file and should never be NULL.